srcpkgs/gf2x/template

```diff
@@ -0,0 +1,34 @@
+# Template file for 'gf2x'
+pkgname=gf2x
+version=1.3.0
+revision=1
+wrksrc=gf2x-gf2x-${version}
+build_style="gnu-configure"
+hostmakedepends="automake libtool"
+short_desc="C/C++ library for fast arithmetic in GF(2)[x]"
+maintainer="Gonzalo Tornaría <tornaria@cmat.edu.uy>"
+license="GPL-3.0-or-later"
+homepage="https://gitlab.inria.fr/gf2x/gf2x"
+distfiles="https://gitlab.inria.fr/gf2x/gf2x/-/archive/gf2x-${version}/gf2x-gf2x-${version}.tar.gz"
+checksum=11bcf98b620c60c2ee3b4460b02b7be741f14cfdc26b542f22c92950926575e0
+
+build_options="native"
```

Comment:
Please add a description for this option, it isn't obvious what it does. `desc_option_native="..."`

I would name it sse3 directly, maybe. Then you could do `configure_args="$(vopt_enable sse3)"`.

Even if you keep the native name, you can still do `configure_args="$(vopt_if native '' '--disable-sse3')"` and have configure args together with build style instead of down here.

Reply:
I added this to `common/options.description` in #29997 (see 2274429). There is a discussion about this in that PR.

My intention is to have an easy way to recompile math software for native arch. Depending on the package sometimes is sse3, sometimes is more. In this case the library also uses sse4.1 and pclmul, but they are dependents so `--disable-sse3` is enough to disable all of that is not generic (x86_64 implies sse2 but nothing more).

Not disabling sse3 as above means that the library will be compiled with -msse3, -msse41, -mpclmul, ... and maybe also -march=native, etc, so that all the options available in the current cpu will be used.

In general, different software have different features, oftentimes they have custom written asm for different architectures so selecting build option "native" will not necessarily be equivalent to just using `-march=native`.

In many cases (like here) one has to be careful because native is the default and it is necessary to make sure the binary actually is generic enough for x86_64 (I'm not sure how to test this -- can a VM disable some sse levels even if the host has them? I have an old core 2 quad (q9550) working (even with 8GB RAM) but that already has sse3!

Reply:
AFAIK you can test, but I'm not sure it's necessary. We can wait for bug reports, IMO.

 Please import the commit into this PR, then :)

Reply:
I wonder if `native_build` would be better.